bestofjs/bestofjs
# Combine projects and tags in the same search results list

## Goal

- Improve the UX when searching, to fix #246 by combining projects and tags in the same search result list, sorted by relevance, instead of separating them.
- When a tag is selected, make the fact that the search is scoped to the tag mroe obvious


## How to test

- Try to search for `react`, notice how the list shows both the React project and the `React` tag
- Open a tag (say the `React` tag), then open the search modal

## Screenshots

#### Default state (no query, no tags)

Showing the most popular tags

![image](https://github.com/bestofjs/bestofjs/assets/5546996/61579e9b-c070-4550-b421-18f4b89319bc)


#### Searching

Showing a mix of projects and tags

![image](https://github.com/bestofjs/bestofjs/assets/5546996/c640517f-3670-43e2-b7e4-a7169d66dcac)

#### Start the search when a tag is selected

- The place holder has changed to explain that the search is now "scoped" to one or several tags
- The list shows the "Related Tags" instead of the popular tags
- The icon of the tag entry has changed to convey the fact that it will open a page that combine several tags (instead of a single one)

![image](https://github.com/bestofjs/bestofjs/assets/5546996/db0f8801-040a-460b-bd55-a172c3d22cb3)

#### Searching when a tag is selected

![image](https://github.com/bestofjs/bestofjs/assets/5546996/dd023742-bd6c-488c-8fc1-ff88815e0949)
